cmd/nerdctl/login.go

```diff
@@ -240,14 +241,44 @@ func loginClientSide(ctx context.Context, cmd *cobra.Command, auth types.AuthCon
 	if len(regHosts) == 0 {
 		return "", fmt.Errorf("got empty []docker.RegistryHost for %q", host)
 	}
+	needFallbackToPlainHttp := false
 	for i, rh := range regHosts {
 		err = tryLoginWithRegHost(ctx, rh)
```

Comment:
```suggestion
                err = tryLoginWithRegHost(ctx, rh)
		if err != nil && insecure && (errutil.IsErrHTTPResponseToHTTPSClient(err) || errutil.IsErrConnectionRefused(err)) {
		    rh.Scheme = "http"
		    err = tryLoginWithReghost(ctx, rh)
		}
```

Reply:
I'm not sure about changing the Scheme directly is the best practice

Reply:
I reserve my opinion. 

When it is not a valid HTTPs request and `insecure-registry` flaged, we can fallback directly to plain HTTP. cc @fahedouch